hydradx/notebooks/Omnipool/vault/ImpermanentLossAnalysis.ipynb

```diff
@@ -0,0 +1,224 @@
+      "text/plain": [
+       "<Figure size 1440x360 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "steps = 101\n",
+    "# price_min = 0.2\n",
+    "# price_max = 5\n",
+    "for price_min, price_max in [(0.2, 5), (0.5, 2)]:\n",
+    "    step_factor = (price_max / price_min) ** (1 / (steps - 1))\n",
+    "    price_move_range = [price_min * step_factor ** i for i in range(steps)]\n",
+    "    events = []\n",
+    "    weight_factors = [0.0001, 0.01, 0.02, 0.035, 0.05]\n",
+    "\n",
+    "\n",
+    "    for w, weight in enumerate(weight_factors):\n",
+    "        assets = {\n",
+    "            'HDX': {'usd price': 0.05, 'weight': 0.8},\n",
+    "            'USD': {'usd price': 1, 'weight': 0.20 - weight},\n",
+    "            'TKN': {'usd price': 1, 'weight': weight}\n",
+    "        }\n",
+    "\n",
+    "        for tkn, info in assets.items():\n",
+    "            liquidity[tkn] = initial_omnipool_tvl * info['weight'] / info['usd price']\n",
+    "            lrna[tkn] = initial_omnipool_tvl * info['weight'] / lrna_price_usd\n",
+    "\n",
+    "        initial_state.pools={\n",
+    "            'omnipool': OmnipoolState(\n",
+    "                tokens={\n",
+    "                    tkn: {'liquidity': liquidity[tkn], 'LRNA': lrna[tkn]} for tkn in assets\n",
+    "                },\n",
+    "                asset_fee=0,\n",
+    "                lrna_fee=0,\n",
+    "            )\n",
+    "        }\n",
+    "\n",
+    "        events.append([])\n",
+    "        for price in price_move_range:\n",
+    "            price_list = [{'TKN': initial_state.external_market['TKN']}, {'TKN': price}]\n",
+    "            initial_state._evolve_function = historical_prices(price_list)\n",
+    "            events[w].append(run.run(initial_state, 1, silent=True)[-1])\n",
+    "\n",
+    "    impermanent_loss = [[\n",
```

Comment:
I think it's better to say that e.g. if the price goes up 90%, the "impermanent loss" is 5% (or -5% depending on sign convention). This variable is dividing the cash_out value by the hold value, so "impermanent_loss" is 95%. Everything is correct but the variable name should probably not be "impermanent_loss" since it's actually the value *retained* for the LP during a price move.

Reply:
That is a good point, I'll change it.